### ISSUE TITLE
Bug when opening Xournal through Windows user interface - Hides pages and tools and/or changes textbox font to brackets font ([][][][]).

### ISSUE THREAD

**Issue by Aic-Ma** [xournalpp/xournalpp#3207]
**Affects versions :**
 - OS: Microsoft Windows 10 Home
 - Which version of libgtk do you use
     - 3.24.23
 - Version of Xournal++:
     - 1.0.20
 - Installation method: 
     - Official page, https://xournalpp.github.io/installation/

**Describe the bug** 
The following bugs may happen, either partially in each case, independently, or simultaneously:

Option 1:
Hides sidebar page review; Hides menu bar options;  Hides main pages.
 
Option 2:
Textbox font is changed to brackets. (eg: << Some textbox text >> might be changed to <<[][][][][][][][][][][][][][][][]>>

**To Reproduce**
1. Step 1: Find Xournal++ from the Windows user interface, or from the Window's search menu.
2. Step 2: Open file

**Expected behavior**
Open Xournal++ without Debug In pop up. Not hide menu bar options nor the sidebar page review nor the main document. Not encode texboxes with bracket notation. 
  
**Screenshots of Problem**
Document 1
![image](https://user-images.githubusercontent.com/86889681/124363972-15e15180-dc04-11eb-94c8-89a63cec4259.png)
![image](https://user-images.githubusercontent.com/86889681/124364266-f0edde00-dc05-11eb-9b27-b0525595d056.png)

Document 2
Expected behaviour:
![image](https://user-images.githubusercontent.com/86889681/124364244-d0be1f00-dc05-11eb-8eda-0b15d8060d80.png)
Variosu Error combinations.
![image](https://user-images.githubusercontent.com/86889681/124364274-02cf8100-dc06-11eb-9f60-e3383a4f8edc.png)
![image](https://user-images.githubusercontent.com/86889681/124364277-08c56200-dc06-11eb-8fdf-ca0eed51c7cc.png)

**A research on the Textbox Bracket Font Bug**
![image](https://user-images.githubusercontent.com/86889681/124366886-ac1f7280-dc18-11eb-962b-f8792dd52d1c.png)

While working on the research 1
![image](https://user-images.githubusercontent.com/86889681/124366585-74173000-dc16-11eb-874b-dbed08bb6e21.png)
While working on the research 2
![image](https://user-images.githubusercontent.com/86889681/124366934-f56fc200-dc18-11eb-869f-efa0caba184b.png)

**Additional context**

- The bugs do not appear when opening a file through Xournal++'s [Recent Documents] option. 
- Interacting with the Journal may alleviate some of these bugs. 
     - On the Hidden Interface Bug
          - Scrolling, writing, erasing may uncover areas of the Journal. 
          -  The menubar options are uncovered when passing mouse over them. 
          - The sidebar layer review can be accessed normally, even when the sidebar page review remains affected. 
     -  On the Textbox Brackets Font Bug
          - Changing text size restores proper text font.  
- Does not appear to be resolved by waiting. Tested waiting for the issues to fix themselves for some 10 minutes. 
- The Debug IN pop up always appears, although I cannot faithfully predict the appearance of the bugs.
- Honorary mention for the Textbox Bracket Error:
     - It affects particular textbox settings used on the document. Can vary each time the file is reopened. 
     - The bug keeps to a given textbox settings each time it appears. Change the textbox settings of a currupted textbox, and it is becomes well. Write again on the given currupted textbox settings, and the font will be of the brackets. 
     - The menubar can be affected. One time, menubar options were afflicted by this bug. Another time, the number on the textbox settings on the menu bar was also affected. The Hidden Interface Bug has in both cases been active.
     - Zooming in and out while having the Textbox Bracket Font Bug active can crash the file.

**Comment by rolandlo** [comment]
Please try [version 1.1.0+dev](https://github.com/xournalpp/xournalpp/tree/nightly) and report if anything of these issues remain.

**Comment by Aic-Ma** [comment]
> Please try [version 1.1.0+dev](https://github.com/xournalpp/xournalpp/tree/nightly) and report if anything of these issues remain.

Hi rolandlo. Tried to run the instructions from the WindowsBuild.md, but miserably failed getting the Build Step to work. Thus, I took the easier road of of using downloading the zip file xournalpp-1.1.0.dev-nightly.20210702-windows.zip from the [Automated nighly build](https://github.com/xournalpp/xournalpp/releases/tag/nightly) page. The problem unfortunately persists. The same as so far applies from the OP, with the exception that the Debug In pop up no longer appears. 

I'm fully conscious the way to go was to follow the steps from the link provided, and that I somehow messed up on a step. Is there not a photo walkthrough on how  to properly compile the setup? 

Thanks for the previous answer, by the way.

**Comment by rolandlo** [comment]
I did not expect you to build Xournal++ from source. You wouldn't get a different result than from the automated nightly build, although of course it would offer more debugging options for you.

No, there is no photo walkthrough on how to compile the setup. If you report the error(s) in the Bulid step we may be able to help.

I'm quite puzzled by the bugs that you report. They have not been reported before, I believe, and I can't reproduce them.
Can you share some info on the computer you are using and what Windows version you use? Is it Windows 10?

**Comment by Aic-Ma** [comment]
@rolandlo, sure, here is some relevant info. I will update the OP to include at least the full OS Name.
Current Laptop:
OS Name:             Microsoft Windows 10 Home
Version:	             10.0.19041 Build 19041
System Model:     13U70P-G.ARW5U1
System Type:	     x64-based PC

But I had another computer, pretty much the same, except some 7 years old, which had the same issues, and I just verified downloading the stable version once again that the bug that changes certain settings to that of brackets is also there. Emphatically, the Debug In pop ups keeps appearing, and I the hidden interface bug after another couple of tries also shows up. 

However, it may be that these are not fully independent cases. When switching to this newer device, I brought over much of my previous Xournal files, all which always mentioned the Debug In pop up. Therefore, I think this problem can also be due to a virus. I may be able to reset this old one as to see if it keeps happening on fresh settings, but I it will have to be not now but in a couple of days. 
Other Laptop:
OS Name:	    Microsoft Windows 10 Home
Version:	            10.0.19043 Build 19043
System Model:     SVP13215CDB
System Type:	     x64-based PC

**Comment by rolandlo** [comment]
The Debug pop ups are to be expected in the stable release (version 1.0.20). The rest is unexpected. Consider uninstalling all Xournal++ versions, removing the contents of `C:\Users\<username>\AppData\Local\xournalpp` (that contains the settings file) and reinstalling Xournal++ 1.1.0+dev (from the zip-archive).

**Comment by Aic-Ma** [comment]
Hi, @rolandlo. I have done the steps twice now, the second time having both deleted my Local/xournalpp folder and my Local  recent-files.xbel file. It's quite interesting to note both time the affected textbox setting after reopening the file has been Sans Normal 12 pt, which has been the settings I have first used on the file before having it saved and closed. As additional context, no other xournal file had been opened, and these were the first files done after having recently installed Xournal++ 1.1.0+dev from the zip-archive. 

I'm running out of ideas here. What do you think we can do?

Thanks for keeping up with me by the way. Here is another picture of the most recent appearance of the bug. 
![image](https://user-images.githubusercontent.com/86889681/124510748-20cdea80-dd9a-11eb-8826-2cc3d6f92106.png)

**Comment by rolandlo** [comment]
`DEFAULT_FONT="Sans" and `DEFAULT_FONT_SIZE=12` are hard-coded in Xournal++, so that part looks normal.
Can you keep the task manager open and observe CPU and memory usage while you get those hidden pieces of the Journal? Seems your computer is lagging behind because it is overworked. Does other software work fluently at the same time?

**Comment by Aic-Ma** [comment]
@rolandlo, sorry I'm running late. I have not extensensively used other programs while having any of the bugs active in Xournal, but I did compared certain aspects, like running some live-script on Matlab, or just openning and closing them, while having the various bugs present. Here is a brief comparison of the computer's performance when the xournal is active presenting any or no bug: [Performance Tests](https://1drv.ms/b/s!Ai3p1sBC2oNUgddEMktLuEKXOIm4Aw). 

It struck me interesting no significant change seems to happen between the control (no bug) and test groups. The timing at Up time is such because more frequently than not I performed first the test with the bug and then the test without the bug. Airplane mode was adopted in the last tests to minimize interference coming from the wifi processes, although that was already unimportant.

By the way, I was surprised to find using the `File -> Recent Documents` option to open the file did brought the file with bugs in two occasions.

**Comment by couldNOTcompute** [comment]
General Information:
    OS: Microsoft Windows 10 Professional
    Hardware: i7-1165G7, 15,8 GB, NVMe SSD (as performance was put forward as a possible cause)
    Version of Xournal++:
        1.1.0
    Installation method:
        Official page

Describe the bug:
1. Textbox font is changed to brackets. [][][][][][][][][][][][][][][][]
2. (While trying to reproduce the first issue) When I open a file the window itself now has the bracket font, the textboxes inside are unaffected. This Issue varies quite a bit! I tried rebooting, upon first trying it did not produce the bracket font, but instead just the mix of English and German. Closing both windows and just reopening the preexisting file the brackets reappear (all 5 times I tried), opening another file just reverts back to the language mix.

To Reproduce:
1. I was unable to find a pattern in the appearance of the issue. Hence, I can't provide a guide to reproduce.
2. There too seems to be no real pattern. At first, I was able to reproduce the issue this way: (Language German) I have one window open an now open a second instance, by clicking a existing file, now the second Instance has the issue while the first is unaffected. Closing and opening the lather window fixes the problem, but now the menus are a strange mix of the en and ger versions.

Expected behavior
Xournalpp should keep the font it is set too.

Additional Context:
- resizing the font (to a larger size) and/or changing it removed the problem multiple times
     Saving it after it after it disappeared, resolved the issue (sadly I forgot to safe them as a different file so the affected files are lost)
- I sync the folder to my ubuntu system via git
    Opening the same files, I had issues on windows did not have the issue
- The font I use is 'Arial Normal' Size 6/8
- The first time it appeared was in an pdf export, but I only observed it in the pdf file not in xournalpp (as far as I recall)

![Screenshot (1)](https://user-images.githubusercontent.com/95862384/145414315-684534eb-1eb9-4aaf-af2e-3d9e22f1cae4.png)
![Screenshot (2)](https://user-images.githubusercontent.com/95862384/145414318-ddde41b2-1331-4b23-bbec-01400d07519e.png)

I'm sorry for that my report is a bit incoherent, but I'm pressed for time and just played around with the issue a bit and couldn't find any real pattern. But if you have anything you'd like to test me feel free to ask.
